libraries/rush-lib/src/logic/operations/CacheableOperationPlugin.ts

```diff
@@ -585,13 +590,13 @@ export class CacheableOperationPlugin implements IPhasedCommandPlugin {
     operationMetadataManager: OperationMetadataManager | undefined;
   }): Promise<ProjectBuildCache | undefined> {
     if (!buildCacheContext.projectBuildCache) {
-      const { cacheDisabledReason } = buildCacheContext;
+      const { cacheDisabledReason, operationSettings, projectChangeAnalyzer, getCacheEntryId } =
+        buildCacheContext;
       if (cacheDisabledReason) {
         terminal.writeVerboseLine(cacheDisabledReason);
         return;
       }
 
-      const { operationSettings, projectChangeAnalyzer } = buildCacheContext;
```

Comment:
These were separated because of the early returns, so that the (admittedly negligible) cost of the property accesses doesn't happen in those code paths, but also to keep the meaning of the variables closer to when they are used.